src/net/tcp/listener.rs

```diff
@@ -36,11 +40,17 @@ impl TcpListener {
             let host = world.current_host_mut();
 
             if !addr.ip().is_unspecified() && !addr.ip().is_loopback() {
-                panic!("{addr} is not supported");
+                return Err(Error::new(
+                    ErrorKind::AddrNotAvailable,
+                    "invalid argument - addr not supported",
+                ));
             }
 
             if addr.is_ipv4() != host.addr.is_ipv4() {
```

Comment:
When we do the larger refactor this will be handled, but for now this seems like a panic. You'd never be handling this in your application, it would be a misconfiguration of the sim.